rpc/client.go

```diff
@@ -130,32 +128,10 @@ func waitForRequest[T serde.RequestPayload, U serde.ResponsePayload](rc *RpcClie
 	return res.Payload
 }
 
-func (rc *RpcClient) WaitForObjectiveCompletion(expectedObjectiveId ...protocols.ObjectiveId) error {
-	completed := make(map[protocols.ObjectiveId]bool)
-
-	for receivedObjectiveId := range rc.CompletedObjectives() {
-		isObjectiveExpected := false
-		for _, expectedObjectiveId := range expectedObjectiveId {
-			if receivedObjectiveId == expectedObjectiveId {
-				isObjectiveExpected = true
-			}
-		}
-		if !isObjectiveExpected {
-			return fmt.Errorf("received unexpected objective completion notification for objective %v", receivedObjectiveId)
-		}
-
-		completed[receivedObjectiveId] = true
-		done := true
-		for _, expectedObjectiveId := range expectedObjectiveId {
-			if !completed[expectedObjectiveId] {
-				done = false
-			}
-		}
-		if done {
-			return nil
-		}
-	}
-	return nil
+// ObjectiveCompleteChan returns a chan that receives an empty struct when the objective with given id is completed
+func (rc *RpcClient) ObjectiveCompleteChan(id protocols.ObjectiveId) <-chan struct{} {
```

Comment:
I wonder if we should rename this to avoid using `chan`, which is golang specific?

Reply:
I think it is OK to be golang specific, here, since this is the golang RPC client. 

When we come to write a JS RPC client, it would likely have a different method set.